tests/main/preseed-core20/task.yaml

```diff
@@ -0,0 +1,162 @@
+summary: Check that preseeding of core20 works.
+
+details: |
+  This test checks that preseeding of UC20 image with snap-preseed
+  command works.
+
+systems: [ubuntu-2*]
+
+environment:
+  PREPARE_IMAGE_DIR: uc20image
+  STORE_ADDR: localhost:11028
+  STORE_DIR: $(pwd)/fake-store-blobdir
+
+prepare: |
+  #shellcheck source=tests/lib/prepare.sh
+  . "$TESTSLIB"/prepare.sh
+  snap download "--channel=${SNAPD_CHANNEL}" snapd
+  repack_snapd_snap_with_deb_content_and_run_mode_firstboot_tweaks /tmp
+
+  mkdir -p "$STORE_DIR/asserts"
+  echo Expose the needed assertions through the fakestore
+  cp "$TESTSLIB"/assertions/developer1.account "$STORE_DIR/asserts"
+  cp "$TESTSLIB"/assertions/developer1.account-key "$STORE_DIR/asserts"
+  cp "$TESTSLIB"/assertions/testrootorg-store.account-key "$STORE_DIR/asserts"
+
+  #shellcheck source=tests/lib/store.sh
+  . "$TESTSLIB"/store.sh
+  setup_fake_store "$STORE_DIR"
+
+  # sign the model assertion
+  gendeveloper1model < "$TESTSLIB"/assertions/developer1-20-dangerous.json > model.assertion
+
+  echo "Creating a new key without a password"
+  expect -f create-key.exp
```

Comment:
See my comment to create-key.exp.

Reply:
Maybe a simple `mkdir -p ~/.snap/gnupg` followed by import will suffice?